### ISSUE TITLE
AVS - Custom Device implement/receive Response message

### ISSUE THREAD

**Issue by nagen1** [alexa/alexa-avs-sample-app#1147]
Hi,
I'm robotics engineer, trying to build a custom device which Alexa built in and wondering few things (tried to lookup in SDK didn't find much or may be I missed it). #Clarification
Here the scenario:
A Custom Device that comes with built in Alexa, microphone, speaker and few local actions that need to perform. Let us say, Rasbperry Pi and I connected it to coffee maker to turn it on/off, make cappuccino latte, Tea and etc... 
If user says, "**Alexa, make me cappuccino**" then coffee maker should trigger that activity (which I will be coding based on response message/text/alert that I get from Alexa directive.

- Does Alexa sends any response in text or JSON format along with SpeechSynthesizer?
- can we send custom speech message. Like, "sure, making cappuccino"
- is there any python related libraries that we can use to prototype fast ?

Appreciate it.

**Comment by sanjayrd** [comment]
Hi @nagen1,

I think what you're looking for is adding your own custom Alexa capabilities by writing your own skill: See https://developer.amazon.com/alexa-skills-kit for more info.

You can build and host skills using AWS Lambda for free (which can be written in Python).

Thanks,
Sanjay

**Comment by nagen1** [comment]
@sanjayrd Not really though. That adds another layer of communication/words and already looked up those skills kit in which I am not much interested into now. 
let me put it in different prospective:
let us say, we have a recycle trash pick robot and I say, "hey bot 101, please pick all milk bottles those are plastic". 
To solve this if we use skill, it's going to be "Alexa, tell 101 to pick all milk bottles" than just directly talking to bot.

more complex: "hey bot 101, please order pizza for me" in this case I don't need extra device with me to do 2 jobs.

**Comment by sanjayrd** [comment]
@nagen1,

Yep, that's an issue with Alexa Skills that there's no solution around currently via Alexa.

The alternative would be to have your own Speech Recognition software running locally and intercept any audio that you want to act on (such as making cappuccino) and send this request to your own software component rather than to the Alexa APIs.

Hope that helps,
Sanjay
